test/test_gpu3.cpp

```diff
@@ -8694,6 +8694,45 @@ TEST_F(NVFuserTest, ChainProjectionToPersistentProducer) {
   auto cg_outputs = fe.runFusion(inputs);
   testValidate(fusion, cg_outputs, inputs, {t5, t8, t11}, __LINE__, __FILE__);
 }
+
+// Test that 3D reductions with broadcasts as the inner-most non-reduction
+// dimension are successfully scheduled.
+// See https://github.com/NVIDIA/Fuser/issues/1471
+TEST_F(NVFuserTest, Reduction3DWithBroadcast) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  auto tv0 = TensorViewBuilder()
+                 .dtype(DataType::Double)
+                 .contiguity({true, true, true, std::nullopt})
+                 .shape({-1, -1, -1, 1})
+                 .build();
+  fusion->addInput(tv0);
+  auto tv1 = sum(tv0, {2, 0});
+  fusion->addOutput(tv1);
+
+  auto options = at::TensorOptions().dtype(at::kDouble).device(at::kCUDA, 0);
+  auto t0 = at::randn({8, 7, 5, 1}, options);
+  std::vector<c10::IValue> inputs({t0});
+
+  std::shared_ptr<ReductionParams> reduction_params =
+      getReductionHeuristics(fusion, inputs);
+  NVF_CHECK(reduction_params, "Reduction heuristic failed!");
+  scheduleReduction(fusion, *reduction_params);
+
+  FusionExecutor fe;
+  fe.compileFusion(fusion, inputs, reduction_params->lparams);
+  auto cg_outputs = fe.runFusion(inputs, reduction_params->lparams);
+
+  // NOTE: here we manually compute a reference tensor. This is to avoid an
+  // error in getReductionSize() which finds the rfactor tensorview which has a
+  // partial sum over a dimension whose size is a block dim.
+  auto ref = at::sum(t0, {2, 0});
+  ASSERT_TRUE(at::allclose(ref, cg_outputs.at(0)));
+  //testValidate(fusion, cg_outputs, inputs, __LINE__, __FILE__);
```

Comment:
This was surprising.
> C++ exception with description "inferred_extent.hasValue() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/validator_utils.h":228, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Couldn't figure out what the dimensions of a tensorview is in evaluation for validation. rS39{( ceilDiv(( ceilDiv(( ceilDiv(( i0 * i2 ), blockDim.x) ), 1)), gridDim.x) )}rf in T4_l[ iblockIdx.z31{gridDim.z}, iS32{( ceilDiv(( ceilDiv(i1, 2) ), gridDim.z) )}, rS39{( ceilDiv(( ceilDiv(( ceilDiv(( i0 * i2 ), blockDim.x) ), 1) ), gridDim.x) )}rf, iS30{2}, bblockIdx.y28{1}, iblockIdx.x38{gridDim.x}rf, ithreadIdx.x35{blockDim.x}rf, rUS37{1}rf ] ca_pos( 2 ) produce_pos( 3 )                                                                           
Exception raised from getReductionSize at /opt/pytorch/nvfuser/csrc/validator_utils.h:228 (most recent call first):

Reply:
The problematic ID is `rS39{( ceilDiv(( ceilDiv(( ceilDiv(( i0 * i2 ), blockDim.x) ), 1) ), gridDim.x) )}rf`.

Reply:
I'm seeing the validation is still commented out, does that mean the example still won't run?

Reply:
It runs, it's just that the validation code cannot determine what tolerance to use. That's because the rfactored sum has a sum axis with an extent that's a function of a grid dim, and the evaluator code doesn't bind launch params when evaluating sizes. I tried passing my own tolerances but I think it still tried to evaluate those extents and hit the same error. I can remove the comment and leave the current check as it's sufficient, and default allclose tols work here.

Reply:
We could probably just modify testValidate to accept an optional LaunchParams which it will bind to its ExprEvaluator if given. I'll give that a shot later and if it's a small change include it. Otherwise can split into another PR.

Reply:
Actually just adding the `lparams` to the `testValidate` call gets us further, but we still hit https://github.com/NVIDIA/Fuser/blob/main/csrc/ir/nodes.cpp#L1455-L1457. Instead, I'm now just copying the un-scheduled Fusion and using that at validation.